### ISSUE TITLE
Cancel button for people, which uses session[:return_to], is broken

### ISSUE THREAD

**Issue by david135** [ReadyResponder/ReadyResponder#123]
in `app/views/people/edit.rb` and elsewhere:

```
<%= link_to 'Cancel', session[:return_to], :class => 'btn' %></br>
```

But `session[:return_to]` isn't set properly when going from http://localhost:3000/people/everybody and clicking the edit button.

See: http://stackoverflow.com/questions/2139996/how-to-redirect-to-previous-page-in-ruby-on-rails, where they write:

> In your edit action, store the requesting url in the session hash, which is available across multiple requests:
> `session[:return_to] ||= request.referer`
> Then redirect to it in your update action, after a successful save:
> `redirect_to session.delete(:return_to)`

It seems that it is never deleted, and so it is never re-set.  See people_controller.rb:

```
  def set_return_path
    session[:return_to] ||= request.referer
  end
```

We set it as recommended, but we never clear the value, so whenever it is set once, it sticks and is never updated. (Until we reset the cookie signing, this had been going to a random, unrelated page, but now it seems to be set to the root page, because that is the first and only time the value is null, and therefore set when we go through `PeopleController` the first time.)

This seems to go back to commit 7572cc7f89c84fba4045a2cf80ef4555a92fd0a7.

**Comment by kgf** [comment]
@david135 In your research, what would be the solution to this ?

**Comment by david135** [comment]
It seems that the idiom is to delete `session[:return_to]` every time we go to it, then leaving it ready to be re-set.  Having the views (edit, new, show) link to `session[:return_to]` makes it a little complicated. I would use a helping function.  If we compare `view/items/new.html.erb` and `view/people/new.html.erb`, we see the difference in behavior.  Do we want this functionality for all models ('cancel' returns us to wherever we were when we selected 'New (Person|Item|Event)')? If so, abstracting that behavior out of the people_controller may be the way to go.

**Comment by david135** [comment]
The cancel button is now working, since the above-mentioned PRs, but this is still a problem in the controller, setting and using `session[:return_to]`, which impacts redirects.

**Comment by kgf** [comment]
@david135 should we close this issue, and open a new one that is more specific ?

**Comment by david135** [comment]
Resolved in PR #197.
